dietpi/dietpi-software

```diff
@@ -12318,6 +12325,27 @@ _EOF_
 			Download_Install "$(curl -sSfL 'https://api.github.com/repos/rclone/rclone/releases/latest' | mawk -F\" "/\"browser_download_url\": .*\/rclone-v[^\"\/]*linux-$arch.deb\"/{print \$4}")"
 
 		fi
+		
+		software_id=206 # Homer
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 ))
+		then
+			Banner_Installing
+		
+			# Backup existing instance
+			if [[ -d '/var/www/homer' ]]
+			then
+				G_DIETPI-NOTIFY 2 'Existing Homer instance found, backing up to /mnt/dietpi_userdata/homer_backup ...'
+				[[ -d '/mnt/dietpi_userdata/homer_backup' ]] || G_EXEC mkdir /mnt/dietpi_userdata/homer_backup
+				[[ -d '/mnt/dietpi_userdata/homer_backup/resources' ]] && G_EXEC rm -Rf /mnt/dietpi_userdata/homer_backup/{workbox-*.js,resources}
+				G_EXEC cp -a /var/www/homer/. /mnt/dietpi_userdata/homer_backup/
+				G_DIETPI-NOTIFY 2 'Removing old assets with hash in file name ...'
+				G_EXEC rm -Rf /var/www/homer/{workbox-*.js,resources}
```

Comment:
Why not removing the old assets before creating the backup? This way we would need to remove them once?

Reply:
Right, but then the backup wouldn't contain any of those assets. It is mainly to restore those assets/configs users might edit, but its another bonus that the backup is a complete one, just in case the download fails.

Better would be actually to download to tmpfs first, and do all the backup creation and install afterwards, so in case of failing download the untouched old instance is still in place 🤔.

Reply:
hmm but we do the same removal on backup as well as on live folder, isn't it`?

```
rm -Rf /mnt/dietpi_userdata/homer_backup/{workbox-*.js,resources}
rm -Rf /var/www/homer/{workbox-*.js,resources}
```

means we could remove them before already? Or do I'm missing something?

Reply:
> hmm but we do the same removal on backup as well as on live folder, isn't it`?

Jep, but the assets from the active instance are then backed up, together with the new `index.html` which loads these assets.